### DISCUSSION TITLE
Dual extruder START_PRINT help please

### DISCUSSION THREAD

**Discussion by vNate007** [supermerill/SuperSlicer#2609] [Q&A]
Hi guys

Moved to SuperSlicer a couple of weeks ago and loving it. Great work guys!

Racking my brain here;  I have a dual extruder printer and probably trying to over-complicate my start_print script. High level; I want to pass parameters from SuperSlicer via the Start G-Code to Klipper and do a probably unnecessary amount of code to figure out if I am actually printing with both nozzles or just either of the two.

First extruder and bed I fgured out : 
```START_PRINT EXTRUDER={first_layer_temperature} BED=[first_layer_bed_temperature]```

But SuperSlicer defines temperature in the filament Settings and first_layer_temperature and first_layer_bed_temperature is the same for all of my filament presets. How do I pass the temperature for the second extruder to my start script?

Or am I being thick?

**Comment by supermerill** [comment]
> How do I pass the temperature for the second extruder to my start script?

`{first_layer_temperature[1]}`

but it's the second extruder, not sure it's used in the print.

**Reply by vNate007** [reply]
Thanks Merill!

That simple? So the parameters for whatever I have selected on the Gcode view in for Extruder 1 will be temperature, first_layer_bed_temperature etc and the parameters for the filament selected for Extruder 2 will be temperature[1], first_layer_bed_temperature[1] etc. 

 I feel dumb. 

Is there a complete list of parameters somewhere? I have looked -  I think - through all the GitHub documentation and could not find it. I saw reference to [initial_extruder] somewhere but cannot find it in the tool-tips. 

Thanks again!

**Reply by supermerill** [reply]
> Is there a complete list of parameters somewhere?

https://github.com/supermerill/SuperSlicer/wiki/Macro-&-Variable-list
https://help.prusa3d.com/en/article/macros_1775

not completely up-to-date.
and some custom gcode has their specific ones explained in their tooltip.

**Reply by vNate007** [reply]
You rock Merill! 
Thank you.
